### PR TITLE
Walk-through: Avoid creating temporary files

### PR DESCRIPTION
**Pull request by chrmarti**
The third change introduces another custom URI scheme for embedded code snippets. The main problem is that the TypeScript extension only supports the file:// scheme. (Another problem is that the markdown needs to be parsed multiple times.)

### CODE REVIEW

**Comment by bpasero** [comment]
@chrmarti looks a LOT better 👍

File: src/vs/workbench/parts/walkThrough/electron-browser/walkThrough.contribution.ts [hunk 1]
```diff
@@ -5,15 +5,17 @@
...
 import { EditorWalkThroughAction } from 'vs/workbench/parts/walkThrough/electron-browser/editor/editorWalkThrough';
 import { Registry } from 'vs/platform/platform';
 import { EditorDescriptor } from 'vs/workbench/browser/parts/editor/baseEditor';
 import { IEditorRegistry, Extensions as EditorExtensions } from 'vs/workbench/common/editor';
 import { SyncDescriptor } from 'vs/platform/instantiation/common/descriptors';
 import { IWorkbenchActionRegistry, Extensions } from 'vs/workbench/common/actionRegistry';
 import { SyncActionDescriptor } from 'vs/platform/actions/common/actions';
+import { IWorkbenchContributionsRegistry, Extensions as WorkbenchExtensions } from 'vs/workbench/common/contributions';
 
 (<IEditorRegistry>Registry.as(EditorExtensions.Editors)).registerEditor(new EditorDescriptor(WalkThroughPart.ID,
```
Comment (bpasero) [inline src/vs/workbench/parts/walkThrough/electron-browser/walkThrough.contribution.ts]: `Registry.as<IEditorRegistry>(EditorExtensions.Editors)`

File: src/vs/workbench/parts/walkThrough/node/walkThroughInput.ts [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+	private name: string;
+	private description: string;
+
+	constructor(
+		name: string,
+		description: string,
+		resource: URI,
+		public readonly onReady: (container: HTMLElement) => void,
+		@IFileService private fileService: IFileService,
```
Comment (bpasero) [inline src/vs/workbench/parts/walkThrough/node/walkThroughInput.ts]: Unused

**Review by bpasero — APPROVED** [review]

**Review by mjbvz — APPROVED** [review]

**Comment by chrmarti** [comment]
Will merge an updated set of changes, thanks!

**Pull request closed without merging**